jkube-kit/config/image/src/main/java/org/eclipse/jkube/kit/config/image/ImageName.java

```diff
@@ -322,6 +321,31 @@ private void parseComponentsBeforeTag(String rest) {
         }
     }
 
+    private String extractImageName(String repository) {
+        int atIndex = repository.lastIndexOf('@');
+        int colonIndex = repository.lastIndexOf(':');
+
+        if (atIndex != -1 && (colonIndex == -1 || atIndex > colonIndex)) {
+            // Find the index of "sha256:" or "sha512:" after the "@" symbol
+            int sha256Index = repository.indexOf("sha256:", atIndex);
+            int sha512Index = repository.indexOf("sha512:", atIndex);
+
+            // Choose the minimum index among "sha256:" and "sha512:"
+            int digestStartIndex = Math.min(sha256Index != -1 ? sha256Index : Integer.MAX_VALUE,
+                sha512Index != -1 ? sha512Index : Integer.MAX_VALUE);
+
+            if (digestStartIndex != -1 && digestStartIndex < colonIndex) {
```

Comment:
Tihs would always evaluate to `true`
```suggestion
            if (digestStartIndex < colonIndex) {
```